feat: added translation rw2 translation for gauges

jmichalek132 opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

Draft starting the work on adding support for remote write 2.0 in the translation package.
Adding support for translating gauges.

This is first iteration and to keep the PR small
* we don't handle duplicate metrics
* only support gauges
* don't handle other labels than metric name
* don't handle exemplars
*  don't handle metadata

<!-- Issue number (e.g. #1234) or full URL to issue, if applicable. -->
#### Link to tracking issue #33661
Fixes

<!--Describe what testing was performed and which tests were added.-->
#### Testing

<!--Describe the documentation added.-->
#### Documentation

<!--Please delete paragraphs that you did not use before submitting.-->

atoulme committed:
[chore] update sapm-proto to 0.16.0 (#35630)

Fixes
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35626

renovate[bot] committed:
Update module github.com/DataDog/datadog-api-client-go/v2 to v2.31.0 (#35678)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[github.com/DataDog/datadog-api-client-go/v2](https://redirect.github.com/DataDog/datadog-api-client-go)
| `v2.30.0` -> `v2.31.0` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fDataDog%2fdatadog-api-client-go%2fv2/v2.31.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fDataDog%2fdatadog-api-client-go%2fv2/v2.31.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fDataDog%2fdatadog-api-client-go%2fv2/v2.30.0/v2.31.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fDataDog%2fdatadog-api-client-go%2fv2/v2.30.0/v2.31.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>DataDog/datadog-api-client-go
(github.com/DataDog/datadog-api-client-go/v2)</summary>

###
[`v2.31.0`](https://redirect.github.com/DataDog/datadog-api-client-go/releases/tag/v2.31.0)

[Compare
Source](https://redirect.github.com/DataDog/datadog-api-client-go/compare/v2.30.0...v2.31.0)

<!-- Release notes generated using configuration in .github/release.yml
at v2.31.0 -->

#### What's Changed

##### Fixed

- change schema used in FastlyServicesResponse by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2700](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2700)

##### Added

- Add new synthetics HTTP javascript assertion by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2616](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2616)
- Dashboards - Toplist widget style - Add palette by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2668](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2668)
- Allow Table Widget requests to specify text replace formatting in
dashboards by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2669](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2669)
- Add documentation for Data Jobs Monitoring summary keys by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2672](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2672)
- Update estimate docs with realtime changes by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2704](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2704)
- Ensure clients can handle empty oneOf objects by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2702](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2702)
- Add referenceTables field to security monitoring endpoints by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2697](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2697)
- Add UA documentation for new DJM usage_type by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2698](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2698)
- Add v2 endpoints for MS Teams Integration by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2707](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2707)
- Add documention for OCI Integration by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2713](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2713)
- Add schema for mobile test by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2682](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2682)
- Add Synthetics endpoint to fetch uptimes in API spec by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2661](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2661)

##### Changed

- Split the synthetics request port field into a oneOf by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2678](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2678)
- Remove unused field `color` in `TeamUpdateAttributes` by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2674](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2674)
- Powerpack add support for prefix and available values by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2683](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2683)
- bump go version to 1.22 by
[@&#8203;amaskara-dd](https://redirect.github.com/amaskara-dd) in
[https://github.com/DataDog/datadog-api-client-go/pull/2692](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2692)
- Update v2 metrics list endpoint filter by metric type to use metric
type category by
[@&#8203;api-clients-generation-pipeline](https://redirect.github.com/api-clients-generation-pipeline)
in
[https://github.com/DataDog/datadog-api-client-go/pull/2705](https://redirect.github.com/DataDog/datadog-api-client-go/pull/2705)

**Full Changelog**:
https://github.com/DataDog/datadog-api-client-go/compare/v2.30.0...v2.31.0

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>

jmichalek132 committed:
feat: added translation rw2 translation for gauges

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

djaglowski committed:
[chore] Update otel, fix tests (#35701)

Signed-off-by: Dan Jaglowski <jaglows3@gmail.com>

dashpole commented on pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go:
```diff
@@ -0,0 +1,122 @@
+	return out, errs
+}
+
+// prometheusConverter converts from OTel write format to Prometheus write format.
```
```suggestion
// prometheusConverterV2 converts from OTLP to Prometheus write 2.0 format.
```

jmichalek132 committed:
chore: fix issues from linter

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

dashpole commented on pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go:
```diff
@@ -0,0 +1,27 @@
+	tsMap, err := FromMetricsV2(payload.Metrics(), settings)
+	require.NoError(t, err)
+	require.NotNil(t, tsMap)
+	// TODO add assert equal with expected values
```
Dropping a comment to make sure we don't forget.  This test should be done before we submit the PR

jmichalek132 replied:
Done

dashpole commented on pkg/translator/prometheusremotewrite/number_data_points_v2_test.go:
```diff
@@ -0,0 +1,63 @@
+		want   func() map[uint64]*writev2.TimeSeries
+	}{
+		{
+			name: "gauge",
```
please add a case for `pt.Flags().NoRecordedValue()`, and for a float gauge.

jmichalek132 replied:
for the second one do you mean DoubleGauge as here https://github.com/jmichalek132/opentelemetry-collector-contrib/blob/11396751f11905fe62df105f444cef9ab531def0/pkg/translator/prometheusremotewrite/testutils_test.go#L230  ?

dashpole replied:
yes, double in OTel means float64 in Go.

jmichalek132 replied:
done

jmichalek132 committed:
chore: add test case showing we don't handle duplicates

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: ran make fmt

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
Update pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

Co-authored-by: David Ashpole <dashpole@google.com>

jmichalek132 commented:
Do we add changelog entry for this PR too?
Do we want this partial PRs to show up in changelog?

renovate[bot] committed:
Update module github.com/antchfx/xpath to v1.3.2 (#35662)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [github.com/antchfx/xpath](https://redirect.github.com/antchfx/xpath)
| `v1.3.1` -> `v1.3.2` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fantchfx%2fxpath/v1.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fantchfx%2fxpath/v1.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fantchfx%2fxpath/v1.3.1/v1.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fantchfx%2fxpath/v1.3.1/v1.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>antchfx/xpath (github.com/antchfx/xpath)</summary>

###
[`v1.3.2`](https://redirect.github.com/antchfx/xpath/releases/tag/v1.3.2)

[Compare
Source](https://redirect.github.com/antchfx/xpath/compare/v1.3.1...v1.3.2)

New Features:

- Supports Unicode chars for Non-English (PR
[#&#8203;100](https://redirect.github.com/antchfx/xpath/issues/100))

Bug Fixed:

-   [#&#8203;101](https://redirect.github.com/antchfx/xpath/issues/101)
-   [#&#8203;102](https://redirect.github.com/antchfx/xpath/issues/102)
-   [#&#8203;104](https://redirect.github.com/antchfx/xpath/issues/104)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>
Co-authored-by: Yang Song <songy23@users.noreply.github.com>

dashpole commented:
Yes, please add a changelog for this PR, and future PRs if there are any notable changes.

jmichalek132 committed:
chore: added test case for nan value

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

renovate[bot] committed:
Update module github.com/open-telemetry/opentelemetry-collector-contrib/internal/common to v0.111.0 (#35688)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[github.com/open-telemetry/opentelemetry-collector-contrib/internal/common](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib)
| `v0.110.0` -> `v0.111.0` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fopen-telemetry%2fopentelemetry-collector-contrib%2finternal%2fcommon/v0.111.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fopen-telemetry%2fopentelemetry-collector-contrib%2finternal%2fcommon/v0.111.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fopen-telemetry%2fopentelemetry-collector-contrib%2finternal%2fcommon/v0.110.0/v0.111.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fopen-telemetry%2fopentelemetry-collector-contrib%2finternal%2fcommon/v0.110.0/v0.111.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>open-telemetry/opentelemetry-collector-contrib
(github.com/open-telemetry/opentelemetry-collector-contrib/internal/common)</summary>

###
[`v0.111.0`](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/blob/HEAD/CHANGELOG.md#v01110)

[Compare
Source](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/compare/v0.110.0...v0.111.0)

##### 🛑 Breaking changes 🛑

- `instanaexporter`: Remove deprecated instanaexporter
([#&#8203;35367](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35367))
Use the `otlp` exporter instead as explained in IBM's
[documentation](https://www.ibm.com/docs/en/instana-observability/current?topic=opentelemetry-sending-data-instana-backend).
- `elasticsearchexporter`: Drop cumulative temporality histogram and
exponential histogram
([#&#8203;35442](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35442))
Cumulative temporality histogram and exponential histogram are not
supported by Elasticsearch. Use cumulativetodeltaprocessor to convert
cumulative temporality to delta temporality.
- `elasticsearchexporter`: Implement receiver-based routing under
\*\_dynamic_index config
([#&#8203;34246](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34246))
- `config`: Move component.UseLocalHostAsDefaultHost feature gate to
stable.
([#&#8203;35569](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35569))
- `metricsgenerationprocessor`: Generate metrics even when the second
metric's value is 0
([#&#8203;35533](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35533))
- `signalfxexporter`: Do not exclude the metric
`container.memory.working_set`
([#&#8203;35475](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35475))
- `sqlqueryreceiver`: Fail if value for log column in result set is
missing, collect errors
([#&#8203;35068](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35068))
- `windowseventlogreceiver`: The 'raw' flag no longer suppresses
rendering info.
([#&#8203;34720](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34720))
    Previously, this flag controlled two behaviors simultaneously:
1. Whether or not the body of the log record was an XML string or
structured object.
    2.  Whether or not rendering info was resolved.
A separate 'suppress_rendering_info' option now controls rendering info
resolution.
This is considered a breaking change because users setting only the
'raw' flag without also setting the
new 'suppress_rendering_info' flag may see a performance decrease along
with more detailed events.

##### 🚩 Deprecations 🚩

- `sapmreceiver`: `access_token_passthrough` is deprecated
([#&#8203;35330](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35330))
    -   "`access_token_passthrough` is deprecated."
- "Please enable include_metadata in the receiver and add the following
config to the batch processor:"
    ```yaml
    batch:
      metadata_keys: [X-Sf-Token]
    ```

##### 🚀 New components 🚀

- `receiver/prometheusremotewrite`: Add a new receiver for Prometheus
Remote Write.

([#&#8203;33782](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/33782))

##### 💡 Enhancements 💡

- `sumconnector`: adds connector and summing logic along with tests
([#&#8203;32669](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/32669))

- `receivercreator`: Validate endpoint's configuration before starting
receivers
([#&#8203;33145](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/33145))

- `otelarrowreceiver`: Add admission control in the otelarrow receiver's
standard otlp data path.
    Also moves admission control config options to a separate block.
    arrow.admission_limit_mib -> admission.request_limit_mib
    arrow.waiter_limit -> admission.waiter_limit

([#&#8203;35021](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35021))

- `clickhouseexporter`: Upgrading stability for traces to beta
([#&#8203;35186](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35186))
The trace exporter has proven to be stable for production deployments.
    Trace configuration is unlikely to receive any significant changes.

- `clickhouseexporter`: Updated the default trace table
([#&#8203;34245](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34245))
    Reduced data types, improved partitioning and time range queries

- `opampsupervisor`: Add configurable logging for the supervisor.
([#&#8203;35466](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35466))

- `datadogreceiver`: Move receiver's metrics stability to alpha.
([#&#8203;18278](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/18278))

- `datadogreceiver`: Add container id from v0.5 request header
([#&#8203;35345](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35345))

- `elasticsearchexporter`: Implement elasticsearch.mapping.hints
attribute handling for data points in OTel mapping mode
([#&#8203;35479](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35479))
elasticsearch.mapping.hints takes a slice of strings. `_doc_count`
enables emitting `_doc_count` for the document.
`aggregate_metric_double` causes histogram or exponential histogram to
be emitted as aggregate_metric_double.

- `elasticsearchexporter`: Revert TSDB array dimension workaround for
metrics OTel mode
([#&#8203;35291](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35291))
Remove the workaround to stringify array dimensions as the limitation
has been lifted in Elasticsearch v8.16.0.

-   `receiver/statsd`: Add support for aggregating on Host/IP.

([#&#8203;23809](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/23809))

- `opampsupervisor`: Skip executing the collector if no config is
provided
([#&#8203;33680](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/33680))

- `googlecloudmonitoringreceiver`: Move receiver's stability to alpha.
([#&#8203;33762](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/33762))

- `hostmetricsreceiver`: Add ability to mute all errors (mainly due to
access rights) coming from process scraper of the hostmetricsreceiver
([#&#8203;20435](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/20435))

- `kubeletstats`: Introduce feature gate for deprecation of
container.cpu.utilization, k8s.pod.cpu.utilization and
k8s.node.cpu.utilization metrics
([#&#8203;35139](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35139))

- `opampsupervisor`: Make supervisor runnable as a Windows Service.
([#&#8203;34774](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34774))

- `opampsupervisor`: Add config option for setting the timeout for the
initial bootstrap information retrieval from the agent
([#&#8203;34996](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34996))

- `pkg/ottl`: Add InsertXML Converter
([#&#8203;35436](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35436))

- `pkg/ottl`: Add GetXML Converter
([#&#8203;35462](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35462))

- `pkg/ottl`: Add ToKeyValueString Converter
([#&#8203;35334](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35334))

- `pkg/ottl`: Add RemoveXML Converter
([#&#8203;35301](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35301))

- `geoipprocessor`: No longer return an error when geo metadata is not
found by a provider.
([#&#8203;35047](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35047))

- `sqlserverreceiver`: Add computer name resource attribute to relevant
metrics
([#&#8203;35040](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35040))

- `windowseventlogreceiver`: Add 'suppress_rendering_info' option.
([#&#8203;34720](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34720))
When this flag is enabled, the receiver will not attempt to resolve
rendering info. This can improve performance
    but comes at a cost of losing some details in the event log.

- `windowseventlogreceiver`: Move artificial "remote_server" field to
'attributes\["server.address"]'.
([#&#8203;34720](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/34720))

##### 🧰 Bug fixes 🧰

- `webhookeventreceiver`: Fixed a bug where request bodies containing
newline characters caused the results to split into multiple log entries
([#&#8203;35028](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35028))
- `opampsupervisor`: Only use TLS config when connecting to OpAMP server
if using `wss` or `https` protocols.
([#&#8203;35283](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35283))
- `metricsgenerationprocessor`: Allow metric calculations to be done on
sum metrics
([#&#8203;35428](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35428))
- `sqlqueryreceiver`: Fix reprocessing of logs when tracking_column type
is timestamp
([#&#8203;35194](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35194))
- `windowseventlogreceiver`: While collecting from a remote windows
host, the stanza operator will no longer log "subscription handle is
already open" constantly during successful collection.
([#&#8203;35520](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35520))
- `windowseventlogreceiver`: If collecting from a remote host, the
receiver will stop collecting if the host restarts. This change
resubscribes when the host restarts.
([#&#8203;35175](https://redirect.github.com/open-telemetry/opentelemetry-collector-contrib/issues/35175))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>
Co-authored-by: Alex Boten <223565+codeboten@users.noreply.github.com>

renovate[bot] committed:
Update module github.com/antchfx/xmlquery to v1.4.2 (#35661)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[github.com/antchfx/xmlquery](https://redirect.github.com/antchfx/xmlquery)
| `v1.4.1` -> `v1.4.2` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fantchfx%2fxmlquery/v1.4.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fantchfx%2fxmlquery/v1.4.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fantchfx%2fxmlquery/v1.4.1/v1.4.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fantchfx%2fxmlquery/v1.4.1/v1.4.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>antchfx/xmlquery (github.com/antchfx/xmlquery)</summary>

###
[`v1.4.2`](https://redirect.github.com/antchfx/xmlquery/releases/tag/v1.4.2)

[Compare
Source](https://redirect.github.com/antchfx/xmlquery/compare/v1.4.1...v1.4.2)

update dep package `github.com/antchfx/xpath` from v1.3.1 to v1.3.2

https://github.com/antchfx/xpath/releases/tag/v1.3.2

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>

jmichalek132 committed:
chore: ran make fmt

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: added double_gauge case

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

renovate[bot] committed:
Update module github.com/Azure/azure-sdk-for-go/sdk/azidentity to v1.8.0 (#35709)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[github.com/Azure/azure-sdk-for-go/sdk/azidentity](https://redirect.github.com/Azure/azure-sdk-for-go)
| `v1.7.0` -> `v1.8.0` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fAzure%2fazure-sdk-for-go%2fsdk%2fazidentity/v1.8.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fAzure%2fazure-sdk-for-go%2fsdk%2fazidentity/v1.8.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fAzure%2fazure-sdk-for-go%2fsdk%2fazidentity/v1.7.0/v1.8.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fAzure%2fazure-sdk-for-go%2fsdk%2fazidentity/v1.7.0/v1.8.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>
Co-authored-by: Yang Song <songy23@users.noreply.github.com>

bogdandrutu committed:
[chore] fix make otelcontribcol, require genotelcontribcol (#35707)

Fixes #35631

No changelog since this does not affect anyone outside this repo.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

sfc-gh-kstewart committed:
Suppress errors on EBADF when unlocking files (#35706)

<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

This error is harmless and happens regularly when `delete_after_read` is
set. This is because we acquire the lock right at the start of the
`ReadToEnd` function and then defer the unlock, but that function also
performs the delete. So, by the time it returns and the defer runs the
file descriptor is no longer valid.

<!--Describe what testing was performed and which tests were added.-->
#### Testing

Tested manually on a laptop running MacOS. When using `acquire_fs_locks`
in conjunction with `delete_after_read` I observe this error in the logs
*before applying this change*:

```
2024-10-08T14:30:23.626-0700	error	reader/reader_unix.go:28	Failed to unlock	{"kind": "receiver", "name": "filelog", "data_type": "logs", "component": "fileconsumer", "path": "<redacted>", "error": "bad file descriptor"}
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader.(*Reader).unlockFile
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.111.0/fileconsumer/internal/reader/reader_unix.go:28
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader.(*Reader).ReadToEnd
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.111.0/fileconsumer/internal/reader/reader.go:126
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume.func1
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.111.0/fileconsumer/file.go:160
```

Performing the same test with this changed applied eliminates the error.

---------

Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

mackjmr committed:
[chore] [receiver/flinkmetrics] Use confighttp.NewDefaultClientConfig instead of manually creating struct (#35615)

**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually
creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

rogercoll committed:
[chore] [receiver/elasticsearch] Use NewDefaultClientConfig instead of manual initialization (#35509)

**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->

Similar to
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35452
but for the `elasticsearch` receiver.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457

**Testing:** <Describe what testing was performed and which tests were
added.>

**Documentation:** <Describe the documentation added.>

mackjmr committed:
[chore] [receiver/apache] Use confighttp.NewDefaultClientConfig instead of manually creating struct (#35576)

**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually
creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

renovate[bot] committed:
Update module github.com/jaegertracing/jaeger to v1.62.0 (#35679)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[github.com/jaegertracing/jaeger](https://redirect.github.com/jaegertracing/jaeger)
| `v1.61.0` -> `v1.62.0` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2fjaegertracing%2fjaeger/v1.62.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2fjaegertracing%2fjaeger/v1.62.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2fjaegertracing%2fjaeger/v1.61.0/v1.62.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2fjaegertracing%2fjaeger/v1.61.0/v1.62.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>jaegertracing/jaeger
(github.com/jaegertracing/jaeger)</summary>

###
[`v1.62.0`](https://redirect.github.com/jaegertracing/jaeger/releases/tag/v1.62.0):
/ v2.0.0-rc2

[Compare
Source](https://redirect.github.com/jaegertracing/jaeger/compare/v1.61.0...v1.62.0)

##### Backend Changes

##### ⛔ Breaking Changes

- \[query] change http and tls server configurations to use otel
configurations
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;6023](https://redirect.github.com/jaegertracing/jaeger/pull/6023))
- \[fix]\[spm]: change default metrics namespace to match new default in
spanmetricsconnector
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;6007](https://redirect.github.com/jaegertracing/jaeger/pull/6007))

##### 🐞 Bug fixes, Minor Improvements

- \[grpc storage]: propagate tenant to grpc backend
([@&#8203;frzifus](https://redirect.github.com/frzifus) in
[#&#8203;6030](https://redirect.github.com/jaegertracing/jaeger/pull/6030))
- \[feat] deduplicate spans based on their hashcode
([@&#8203;cdanis](https://redirect.github.com/cdanis) in
[#&#8203;6009](https://redirect.github.com/jaegertracing/jaeger/pull/6009))

##### 🚧 Experimental Features

- \[jaeger-v2] consolidate v1 and v2 configurations for grpc storage
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;6042](https://redirect.github.com/jaegertracing/jaeger/pull/6042))
- \[jaeger-v2] use environment variables in kafka config
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;6028](https://redirect.github.com/jaegertracing/jaeger/pull/6028))
- \[jaeger-v2] align cassandra storage config with otel
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;5949](https://redirect.github.com/jaegertracing/jaeger/pull/5949))
- \[jaeger-v2] refactor configuration for query service
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;5998](https://redirect.github.com/jaegertracing/jaeger/pull/5998))
- \[v2] add temporary expvar extension
([@&#8203;yurishkuro](https://redirect.github.com/yurishkuro) in
[#&#8203;5986](https://redirect.github.com/jaegertracing/jaeger/pull/5986))

##### 👷 CI Improvements

- \[ci] disable fail fast behaviour for ci workflows
([@&#8203;mahadzaryab1](https://redirect.github.com/mahadzaryab1) in
[#&#8203;6052](https://redirect.github.com/jaegertracing/jaeger/pull/6052))
- Testifylint: enable go-require
([@&#8203;mmorel-35](https://redirect.github.com/mmorel-35) in
[#&#8203;5983](https://redirect.github.com/jaegertracing/jaeger/pull/5983))
- Fix regex for publishing v2 image
([@&#8203;yurishkuro](https://redirect.github.com/yurishkuro) in
[#&#8203;5988](https://redirect.github.com/jaegertracing/jaeger/pull/5988))

##### 📊 UI Changes

##### 🐞 Bug fixes, Minor Improvements

- Support uploads of .jsonl files
([@&#8203;Saumya40-codes](https://redirect.github.com/Saumya40-codes) in
[#&#8203;2461](https://redirect.github.com/jaegertracing/jaeger-ui/pull/2461))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>
Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

renovate[bot] committed:
Update dockerfile deps (#35674)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| python | final | minor | `3.12-slim` -> `3.13-slim` |
| redis | final | patch | `7.4.0` -> `7.4.1` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mackjmr committed:
[chore] [receiver/expvar] Use confighttp.NewDefaultClientConfig instead of manually creating struct (#35614)

**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually
creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

jmichalek132 commented:
Not sure why the unit tests ([link](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35703/files#diff-6e1adb2d853fc9fb690c2a477246dff4a90b46d5f277c340973150db20623ec0R116)) keep failing when the expected and the output matches https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11243820134/job/31260663519?pr=35703#step:8:321.

odubajDT committed:
[chore]: fix go routine leaks in tests (#34729)

**Description:** <Describe what has changed.>
- removing re-running parameter from gotestsum
- fixing tests/code with go routine leaks
- resolving race conditions (mostly caused by parallel tests)
- placing goleak ignorers for go routine leaks from external libraries

**Link to tracking Issue:** #34495

---------

Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>
Co-authored-by: Alex Boten <223565+codeboten@users.noreply.github.com>

mackjmr committed:
[chore] [exporter/prometheusremotewrite] Use NewDefaultClientConfig instead of manually creating struct (#35542)

**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually
creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

aabmass committed:
[chore] [googlecloud] add links to GCP observability overview in READMEs (#35568)

**Description:** Link to GCP documentation in the googlecloud and
googlemanagedprometheus exporters

Co-authored-by: David Ashpole <dashpole@google.com>

narcis96 committed:
goldendataset: fix histogram metric BucketCounts length (#35610)

Correct Histograms as [described in
OTLP](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L461)

Link to tracking Issue: Fixes
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35594

Testing: Tests passed

tcolgate committed:
fix(hostmetricsreceiver): do not duplicate mountpoint metrics (#34635)

Mountpoints can be reported multiple times for each mount into a
namespace. This causes duplicate metrics which causes issues with some
exporters. Each instance of the mountpoint will have identical metrics,
so it is safe to ignore repeated mountpoints.

Closes #34512

**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were
added.>

**Documentation:** <Describe the documentation added.>

mauri870 committed:
[chore][exporter/elasticsearch] fix flaky assertion in TestExporterMetrics (#35639)

#### Description

Sometimes the order of fields in the json strings change which causes
failures in the test.

mackjmr committed:
[chore] [exporter/logzio] Use NewDefaultClientConfig instead of manually creating struct (#35524)

**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually
creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

VihasMakwana committed:
[chore][fileconsumer/archive] - add signature methods for archive feature (#35098)

Description:

This PR introduces the `archive` method to store files older than 3 poll
cycles.
The core logic for reading will be introduced in a follow up PR.


Link to tracking Issue:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32727

Testing: To be included in future PRs

Documentation: To be included in future PRs

ChrsMark committed:
[receiver/nginx] Use NewDefaultClientConfig instead of manually creating struct (#35452)

**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
Similar to
https://github.com/open-telemetry/opentelemetry-collector/pull/11273.
This PR makes usage of the `NewDefaultClientConfig` instead of manually
creating the `confighttp.ClientConfig` struct as part of
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457

**Testing:** <Describe what testing was performed and which tests were
added.>

**Documentation:** <Describe the documentation added.>

---------

Signed-off-by: ChrsMark <chrismarkou92@gmail.com>

dashpole commented:
Yeah strange.  Almost seems like it is something to do with the NaNs.

jmichalek132 committed:
chore: double gauge use float value in test

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

colelaven committed:
[chore] Fix: Fix flaky logdedupprocessor tests (#35719)

#### Description
Fixes flaky tests in logdedupprocessor that were failing due to
nondeterministic ordering of log records. Those tests now ignore log
record order when comparing logs.

#### Testing
Ran the fixed tests 200+ times without fail.

renovate[bot] committed:
Update module github.com/tidwall/gjson to v1.18.0 (#35694)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [github.com/tidwall/gjson](https://redirect.github.com/tidwall/gjson)
| `v1.17.3` -> `v1.18.0` |
[![age](https://developer.mend.io/api/mc/badges/age/go/github.com%2ftidwall%2fgjson/v1.18.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/go/github.com%2ftidwall%2fgjson/v1.18.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/go/github.com%2ftidwall%2fgjson/v1.17.3/v1.18.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/go/github.com%2ftidwall%2fgjson/v1.17.3/v1.18.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>tidwall/gjson (github.com/tidwall/gjson)</summary>

###
[`v1.18.0`](https://redirect.github.com/tidwall/gjson/compare/v1.17.3...v1.18.0)

[Compare
Source](https://redirect.github.com/tidwall/gjson/compare/v1.17.3...v1.18.0)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "on tuesday" (UTC), Automerge - At any
time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/open-telemetry/opentelemetry-collector-contrib).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC45Ny4wIiwidXBkYXRlZEluVmVyIjoiMzguOTcuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOlsiZGVwZW5kZW5jaWVzIiwicmVub3ZhdGVib3QiXX0=-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: opentelemetrybot <107717825+opentelemetrybot@users.noreply.github.com>
Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

jmichalek132 committed:
chore: implemented test case for FromMetricsV2

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 commented:
@dashpole could you please add @ArthurSens and @aknuds1 as reviewers too?

dashpole commented:
I can't add @aknuds1, but i've added arthur

aknuds1 commented:
I'm having a look.

jmichalek132 committed:
chore: fix the test case with NaNs not passing

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: ran make fmt

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: added changelog entry

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: fix name of the component

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 committed:
chore: remove discussed comment

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

dashpole commented:
```
prometheusremotewrite/metrics_to_prw_v2_test.go:11: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
prometheusremotewrite/metrics_to_prw_v2_test.go:15: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)

	"github.com/stretchr/testify/require"
prometheusremotewrite/number_data_points_v2_test.go:10: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
```

jmichalek132 committed:
chore: ran make gci

Signed-off-by: Juraj Michalek <juraj.michalek132@gmail.com>

jmichalek132 commented:
> ```
> prometheusremotewrite/metrics_to_prw_v2_test.go:11: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
> 	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
> prometheusremotewrite/metrics_to_prw_v2_test.go:15: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
> 
> 	"github.com/stretchr/testify/require"
> prometheusremotewrite/number_data_points_v2_test.go:10: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
> ```
> 
> looks like imports are not grouped corrrectly

should be fixed.

ArthurSens commented on pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go:
```diff
@@ -0,0 +1,122 @@
+	prometheustranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+)
+
+// FromMetricsV2 converts pmetric.Metrics to Prometheus remote write format 2.0.
+func FromMetricsV2(md pmetric.Metrics, settings Settings) (map[string]*writev2.TimeSeries, error) {
+	c := newPrometheusConverterV2()
+	errs := c.fromMetrics(md, settings)
+	tss := c.timeSeries()
+	out := make(map[string]*writev2.TimeSeries, len(tss))
+	for i := range tss {
+		out[strconv.Itoa(i)] = &tss[i]
+	}
+
+	return out, errs
+}
+
+// prometheusConverterV2 converts from OTLP to Prometheus write 2.0 format.
+type prometheusConverterV2 struct {
+	// TODO handle conflicts
+	unique      map[uint64]*writev2.TimeSeries
+	symbolTable writev2.SymbolsTable
+}
+
+func newPrometheusConverterV2() *prometheusConverterV2 {
+	return &prometheusConverterV2{
+		unique:      map[uint64]*writev2.TimeSeries{},
+		symbolTable: writev2.NewSymbolTable(),
+	}
+}
```
Since prometheusconverter is a private object, users of this package need a way of retrieving the symbols table somehow.

Maybe FromMetricsV2 should also return that?

dashpole replied:
Is the symbol table meant to be per-request, or long-lived?  But regardless it probably makes sense to return both, since both are needed in the request.

ArthurSens replied:
It's per request. PRW is meant to be stateless

dashpole replied:
then returning it makes sense to me.

jmichalek132 replied:
Will update the PR, should we also make sure it's empty before each request?

ArthurSens replied:
FromMetrics() starts with newPrometheusConverterV2(), so I'm assuming it's impossible to not be empty, right?

jmichalek132 replied:
oh yeah your are right

jmichalek132 replied:
Addressed in new PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35734

jmichalek132 committed:
make FromMetricsV2 return symbols table too

jmichalek132 commented:
sorry my bad closing this PR and opening a new one

bertysentry commented:
[like]  Bertrand Martin reacted to your message:
________________________________
From: Juraj Michálek ***@***.***>
Sent: Thursday, October 10, 2024 7:34:04 PM
To: open-telemetry/opentelemetry-collector-contrib ***@***.***>
Cc: Bertrand Martin ***@***.***>; Review requested ***@***.***>
Subject: Re: [open-telemetry/opentelemetry-collector-contrib] feat: added translation rw2 translation for gauges (PR #35703)


@jmichalek132 commented on this pull request.

________________________________

In pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go<https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35703#discussion_r1795985142>:

> +// FromMetricsV2 converts pmetric.Metrics to Prometheus remote write format 2.0.
+func FromMetricsV2(md pmetric.Metrics, settings Settings) (map[string]*writev2.TimeSeries, error) {
+       c := newPrometheusConverterV2()
+       errs := c.fromMetrics(md, settings)
+       tss := c.timeSeries()
+       out := make(map[string]*writev2.TimeSeries, len(tss))
+       for i := range tss {
+               out[strconv.Itoa(i)] = &tss[i]
+       }
+
+       return out, errs
+}
+
+// prometheusConverterV2 converts from OTLP to Prometheus write 2.0 format.
+type prometheusConverterV2 struct {
+       // TODO handle conflicts
+       unique      map[uint64]*writev2.TimeSeries
+       symbolTable writev2.SymbolsTable
+}
+
+func newPrometheusConverterV2() *prometheusConverterV2 {
+       return &prometheusConverterV2{
+               unique:      map[uint64]*writev2.TimeSeries{},
+               symbolTable: writev2.NewSymbolTable(),
+       }
+}


Addressed in new PR #35734<https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35734>

—
Reply to this email directly, view it on GitHub<https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35703#discussion_r1795985142>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHYD3YWKY4OLYHUJQXVXAKDZ23JCZAVCNFSM6AAAAABPS64XRKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGNRRGI2DCNZYG4>.
You are receiving this because your review was requested.Message ID: ***@***.***>
